src/Model/IndexInfo.php

```diff
@@ -100,6 +154,42 @@ public function getNamespace()
         return (string) $this->info['ns'];
     }
 
+    /**
+     * Return the partialFilterExpression.
+     *
+     * @return array|null
+     */
+    public function getPartialFilterExpression()
+    {
+        if ( ! empty($this->info['partialFilterExpression'])) {
+            return $this->info['partialFilterExpression'];
+        }
+    }
+
+    /**
+     * Return the storage engine
+     *
+     * @return string|null
+     */
+    public function getStorageEngineName()
+    {
+        if ( ! empty($this->info['storageEngine'][0])) {
+            return $this->info['storageEngine'][0];
+        }
+    }
+
+    /**
+     * Return the text index version.
+     *
+     * @return integer|null
+     */
+    public function getTextIndexVersion()
+    {
+        if ( ! empty($this->info['textIndexVersion'])) {
+            return $this->info['textIndexVersion'];
```

Comment:
In line 200 (just below), the ``getVersion()`` method does a cast to an ``(integer)`` for the version number. Perhaps you want to do that here, as well as for ``get2dsphereIndexVersion`` too?